submissions/Harnytskyi/friends-app/js/script.js

```diff
@@ -0,0 +1,163 @@
+const FILTER_GENDER = document.getElementById("filter-gender");
+let friends = [];
+let gender = {
+    male: true,
+    female: true
+}
+
+let selectedFriends = [...friends];
+function filterByGender() {
+    if (gender.male == true && gender.female == true)
+        selectedFriends = friends;
+    else if (gender.male == true && gender.female == false)
+        selectedFriends = friends.filter(item => item.gender == "male")
+    else if (gender.male == false && gender.female == true)
+        selectedFriends = friends.filter(item => item.gender == "female")
+    search();
+}
+function displayFriends() {
+    GRID.innerHTML = selectedFriends.map(item => {
+        return `<div id="${item.name}" class="card">
+            <img class="photo" src="${item.photo}">
+            <p class="name">${item.name}</p>
+            <p class="age">Age: ${item.age}</p>
+            <a class="phone" href="tel:${item.phone}">${item.phone}</a>
+            <div class="email">
+                <a href="mailto:${item.email}">
+                    <button class="button-message">SEND MESSAGE</button>
+                </a>
+            </div>
+        </div>`
+    }).join("");
+}
+
+function changeFilterStatus(checkbox) {
+    if (checkbox.checked) {
+        gender[checkbox.value] = true;
+    } else {
+        if (checkbox.value == "male") {
+            if (gender.female == false) {
+                gender.female = true;
+                CHECKBOX_WOMAN.click();
+            }
+        }
+        else if (checkbox.value == "female") {
+            if (gender.male == false) {
+                gender.male = true;
+                CHECKBOX_MAN.click();
+            }
+        }
+        gender[checkbox.value] = false;
+    }
+    filterByGender();
+    displayFriends();
+}
+function searchEvent() {
+    filterByGender();
+    displayFriends();
+}
+
+function search() {
+    let searchRequest = SEARCH_INPUT.value.toUpperCase();
+    searchedFriends = [];
+    selectedFriends.forEach(item => {
+        if (item.name.toUpperCase().includes(searchRequest)) {
+            searchedFriends.push(item)
+        }
+        selectedFriends = searchedFriends;
+    })
+}
+
+function sortFriends(value) {
+    let criterion;
+    let order;
+    switch (value) {
+        case 'alphabet-ascend':
+            criterion = 'name';
+            break;
+        case 'alphabet-descend':
+            criterion = 'name';
+            order = 'descending';
+            break;
+        case 'age-ascend':
+            criterion = 'age';
+            break;
+        case 'age-descend':
+            criterion = 'age';
+            order = 'descending';
+            break;
+    }
+    friends.sort((a, b) => {
+        if (order == "descending") {
+            let c = a;
+            a = b;
+            b = c;
+        }
+        if (a[criterion] > b[criterion]) return 1;
+        if (a[criterion] == b[criterion]) return 0;
+        if (a[criterion] < b[criterion]) return -1;
```

Comment:
Please, go thru other PRs and find inspiration for the simplest implementation of the sorting function.
As an example, you don't need `order`.